### DISCUSSION TITLE
Client for multipart/mixed or multipart/form-data endpoints?

### DISCUSSION THREAD

**Discussion by retrofreak83** [quarkiverse/quarkus-openapi-generator#710] [Q&A]
Dear all,

I am working on a system which needs to communicate with a REST API I have an OpenAPI spec for. One endpoint is responding in `multipart/mixed` (they also provide another one which returns `multipart/form-data`).

The client code is generated fine. However, when the client code is called to access that multipart-endpoint, an exception like this occurs:
```Jakarta.ws.rs.ProcessingException: RESTEASY003145: Unable to find a MessageBodyReader of content-type multipart/form-data:boundary=MULTIPART_BOUNDARY and type class...```

As REST client dependency, we use `quarkus-rest-client-jackson`, so according to these sites:

https://quarkus.io/guides/rest-migration#dependencies-2
https://quarkus.io/guides/rest-client

we are using the current REST client variant of Quarkus, formerly known as "RESTEasy Reactive".
However, the documentation of the quarkus-openapi-generator plugin calls that dependency "RESTEasy Classic"
https://docs.quarkiverse.io/quarkus-openapi-generator/dev/client.html

In contrast to that, Quarkus documentation calls 
`quarkus-resteasy-client` "the REST Client compatible with RESTEasy Classic":
https://quarkus.io/guides/resteasy-client

What surprises me is that in some places in all this documentation it is mentioned how you can write a REST server that offers endpoints with multipart responses. It is also explained how you can *send* a multipart request with a REST client, but there is nothing about *receiving* a multipart-response.

So, I am wondering whether
- multipart client is possible with API code generated by `quarks-openapi-generator` at all?
- if so, which REST client variant would be needed?
And, which additional (glue) code/annotation(s) would be needed. 
While searching for a solution, I found basically information like:
- works automagically
- Annotate API with `@RegisterProvider(MimeMultipartProvider.class)`
- the model class needs to have appropriate `@PartType` and/or `@FormParam` annotations

Any hints are appreciated :)

**Comment by Zitschaey** [comment]
@retrofreak83 Hey, i came acrossthe same Problem, did you find any solution to the problem? :)

**Reply by retrofreak83** [reply]
Short answer: No. What we did is discuss with the provider of the API, since we believe that sending multipart responses is not an optimal solution. There is a newer version of the API now which in fact just don't uses multipart anymore 🤓
But development on that part of the application has been put on hold since May 2024, but may be resumed soon. Maybe this will be still relevant for me, but also maybe not :)
